koku/masu/external/kafka_msg_handler.py

```diff
@@ -154,33 +155,28 @@ def download_payload(request_id, url, context={}):
         shutil.rmtree(temp_dir)
         msg = f"Unable to download file. Error: {str(err)}"
         LOG.warning(log_json(request_id, msg=msg), exc_info=err)
-        raise KafkaMsgHandlerError(msg)
+        raise KafkaMsgHandlerError(msg) from err
 
     sanitized_request_id = re.sub("[^A-Za-z0-9]+", "", request_id)
-    gzip_filename = f"{sanitized_request_id}.tar.gz"
-    temp_file = f"{temp_dir}/{gzip_filename}"
+    temp_file = Path(temp_dir, sanitized_request_id).with_suffix(".tar.gz")
     try:
-        temp_file_hdl = open(temp_file, "wb")
-        temp_file_hdl.write(download_response.content)
-        temp_file_hdl.close()
+        temp_file.write_bytes(download_response.content)
     except OSError as error:
         shutil.rmtree(temp_dir)
         msg = f"Unable to write file. Error: {str(error)}"
         LOG.warning(log_json(request_id, msg=msg, context=context), exc_info=error)
-        raise KafkaMsgHandlerError(msg)
+        raise KafkaMsgHandlerError(msg) from error
 
-    return (temp_dir, temp_file, gzip_filename)
+    return Path(temp_file)
```

Comment:
This is already a `Path` object.
```suggestion
    return temp_file
```